com.microsoft.mrtk.uxcore/Keyboard/NonNative/NonNativeKey.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UX
+{
+    /// <summary>
+    /// Abstract class representing a key in the non native keyboard
+    /// </summary>
+    public abstract class NonNativeKey : MonoBehaviour
+    {
+        /// <summary>
+        /// Reference to the GameObject's button component.
+        /// </summary>
+        [field: SerializeField, Tooltip("Reference to the GameObject's button component.")]
+        protected PressableButton Button { get; set; }
```

Comment:
Should be a `StatefulInteractable`. Should always specify the least-specific type that still implements the necessary functionality

(Useful if someone builds a different sort of button or affordance that will provide key-like functionality)

Reply:
(and change the doccomment, "interactable component")

Reply:
I doubt if people will really build their own button 99.9999% of the time, but modified as requested.